main.go

```diff
@@ -8,32 +8,42 @@ import (
 	"os"
 	"strings"
 
-	"github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"sigs.k8s.io/yaml"
 )
 
-func handleDocument(provider string, document []byte) {
-	var release v1alpha1.Release
-	err := yaml.Unmarshal(document, &release)
+func handleDocument(provider, annotationKey string, document []byte) {
+	var release unstructured.Unstructured
+	err := yaml.UnmarshalStrict(document, &release)
 	if err != nil {
 		fmt.Println(err)
-		os.Exit(4)
+		os.Exit(1)
 	}
 
-	if release.Kind == "Release" {
-		release.Annotations["giantswarm.io/release-notes"] = fmt.Sprintf("https://github.com/giantswarm/releases/tree/master/%s/%s", provider, release.Name)
+	if release.GetKind() == "Release" {
+		annotations := release.GetAnnotations()
+		if annotations == nil {
```

Comment:
We expect this to always have a default value so this is probably fine, but you could also just initialize the map if it's nil.

Reply:
If the map is nil then there are bigger problems in the releases repository I think. So I'll remove this check altogether..